pkg/controller/tas/node_failure_controller.go

```diff
@@ -87,7 +87,12 @@ func (r *nodeFailureReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	}
 	readyCondition := utiltas.GetNodeCondition(&node, corev1.NodeReady)
 	if readyCondition.Status == corev1.ConditionTrue {
-		return ctrl.Result{}, nil
+		affectedWorkloads, err = r.getWorkloadsOnNode(ctx, req.Name)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		patchErr := r.removeNodeToReplaceAnnotation(ctx, req.Name, affectedWorkloads)
+		return ctrl.Result{}, patchErr
```

Comment:
```suggestion
		return ctrl.Result{}, r.removeNodeToReplaceAnnotation(ctx, req.Name, affectedWorkloads)
```

nit: we can directly return this error.